From 7d52945b0b261c54ec736ecc435936c5fb8b81a4 Mon Sep 17 00:00:00 2001
From: Yousong Zhou <yszhou4tech@gmail.com>
Date: Thu, 30 Nov 2017 11:07:12 +0800
Subject: [PATCH] Fix app compilation failure from inclusion of wrong headers

Compiling against glibc can fail with the following errors

     In file included from /home/stefan/Router/o26431-vmmc_v5/source/staging_dir/toolchain-mips_24kc_gcc-5.5.0_glibc/include/bits/fcntl-linux.h:321:0,
		     from /home/stefan/Router/o26431-vmmc_v5/source/staging_dir/toolchain-mips_24kc_gcc-5.5.0_glibc/include/bits/fcntl.h:104,
		     from /home/stefan/Router/o26431-vmmc_v5/source/staging_dir/toolchain-mips_24kc_gcc-5.5.0_glibc/include/fcntl.h:35,
		     from linux/ifxos_linux_device_access_appl.c:30:
    /home/stefan/Router/o26431-vmmc_v5/source/build_dir/target-mips_24kc_glibc/linux-lantiq_xrx200/linux-4.9.65/include/linux/falloc.h:12:2: error: unknown type name '__s16'
      __s16  l_type;
      ^
    /home/stefan/Router/o26431-vmmc_v5/source/build_dir/target-mips_24kc_glibc/linux-lantiq_xrx200/linux-4.9.65/include/linux/falloc.h:13:2: error: unknown type name '__s16'
      __s16  l_whence;
      ^
    /home/stefan/Router/o26431-vmmc_v5/source/build_dir/target-mips_24kc_glibc/linux-lantiq_xrx200/linux-4.9.65/include/linux/falloc.h:14:2: error: unknown type name '__s64'
      __s64  l_start;
      ^
    /home/stefan/Router/o26431-vmmc_v5/source/build_dir/target-mips_24kc_glibc/linux-lantiq_xrx200/linux-4.9.65/include/linux/falloc.h:15:2: error: unknown type name '__s64'
      __s64  l_len;  /* len == 0 means until end of file */
      ^
    /home/stefan/Router/o26431-vmmc_v5/source/build_dir/target-mips_24kc_glibc/linux-lantiq_xrx200/linux-4.9.65/include/linux/falloc.h:16:2: error: unknown type name '__s32'
      __s32  l_sysid;
      ^
    /home/stefan/Router/o26431-vmmc_v5/source/build_dir/target-mips_24kc_glibc/linux-lantiq_xrx200/linux-4.9.65/include/linux/falloc.h:17:2: error: unknown type name '__u32'
      __u32  l_pid;
      ^
    /home/stefan/Router/o26431-vmmc_v5/source/build_dir/target-mips_24kc_glibc/linux-lantiq_xrx200/linux-4.9.65/include/linux/falloc.h:18:2: error: unknown type name '__s32'
      __s32  l_pad[4]; /* reserved area */
      ^
    Makefile:1945: recipe for target 'libifxos_a-ifxos_linux_device_access_appl.o' failed

Ref: https://bugs.openwrt.org/index.php?do=details&task_id=1196
---
 src/Makefile.am | 6 +-----
 1 file changed, 1 insertion(+), 5 deletions(-)

--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -14,11 +14,7 @@ lib_LIBRARIES = libifxos.a
 
 AM_CPPFLAGS = \
 	-I@srcdir@\
-	-I@srcdir@/include \
-	-I@KERNEL_INCL_PATH@ \
-	-I@KERNEL_BUILD_PATH@/include \
-	-I@KERNEL_BUILD_PATH@/include2 \
-	-I$(TARGET_INCL_PATH)
+	-I@srcdir@/include
 
 if HAVE_GCC
 
